rr/config/app_validator.py

```diff
@@ -0,0 +1,260 @@
+
+        filter_labels = filt['labels']
+        validate_lists(
+            filter_labels,
+            str,
+            "Found label element in filter, but it is not a string")
+
+        filters_list.append(filt['name'])
+
+    return filters_list
+
+
+def validate_actions(cfg):
+    """Validates the avtions field of the configuration object
+    """
+
+    validate_objects(
+        cfg,
+        'actions',
+        list,
+        "Actions field not found",
+        "Found actions field, but is is not a list")
+    actions = cfg['actions']
+
+    actions_list = []
+    for action in actions:
+        validate_objects(
+            action,
+            'name',
+            str,
+            "Name field not found in action",
+            "Found name field in action, but it is not a string")
+        validate_objects(
+            action,
+            'type',
+            str,
+            "Type field not found in action",
+            "Found type field in action, but it is not a string")
+        validate_objects(
+            action,
+            'location',
+            str,
+            "Location field not found in action",
+            "Found location field in action, but it is not a string")
+
+        action_type = action['type']
+        if (action_type == 'recording'):
+            validate_objects(
+                action,
+                'lenght',
+                int,
+                "Lenght field not found in action of type recording",
+                "Lenght field in action must be a whole number")
+
+        actions_list.append(action['name'])
+
+    return actions_list
+
+
+def validate_triggers(cfg, actions, filters):
+    """Validates the triggers field of the configuration object
+    """
+
+    validate_objects(
+        cfg,
+        'triggers',
+        list,
+        "Triggers field not found",
+        "Found triggers field, but is is not a list")
+    triggers = cfg['triggers']
+
+    triggers_list = []
+    for trigger in triggers:
+        validate_objects(
+            trigger,
+            'name',
+            str,
+            "Name field not found in trigger",
+            "Found name field in action, but it is not a string")
+        validate_objects(
+            trigger,
+            'action',
+            str,
+            "Action field not found in trigger",
+            "Found action field in trigger, but it is not a string")
+        validate_objects(
+            trigger,
+            'filters',
+            list,
+            "Filters field not found in trigger",
+            "Found filters field in trigger, but it is not a list")
+
+        trigger_filters = trigger['filters']
+        validate_lists(
+            trigger_filters,
+            str,
+            "Found filter element in trigger, but it is not a string")
+
+        for filt in trigger_filters:
+            validate_dependency(
+                filt, filters, "Trigger has a not defined filter")
```

Comment:
It would be helpful if you said which one is not defined. For example `Triggers attempts to use the \"%s\" filter, but it is not defined anywhere`.